File: proposed-projects/round-trip-experiment/reverse/20260909T002206Z_273cf6b9/20260910T042804Z_e72f6c04/ported_implementation/gbnf/utils/is_point_in_range_test.py
import pytest

from .is_point_in_range import is_point_in_range


@pytest.mark.parametrize(
    ("point", "given_range", "expectation"),
    [
        (96, [97, 122], False),
        (97, [97, 122], True),
        (98, [97, 122], True),
        (122, [97, 122], True),
        (123, [97, 122], False),
    ],
)
def test_it_checks_if_point_is_in_range(point, given_range, expectation):
    assert is_point_in_range(point, given_range) is expectation

File: packages/gbnf-experiment/data/20260909T124157Z_4ee67ba2/ported_implementation/tests/unit/utils/is-point-in-range.test.ts
import assert from 'node:assert/strict';
import { describe, test } from 'node:test';
import { isPointInRange } from '../../../src/utils/is-point-in-range.ts';
import type { Range } from '../../../src/index.ts';

const CASES: [number, Range, boolean][] = [
  [96, [97, 122], false],
  [97, [97, 122], true],
  [98, [97, 122], true],
  [122, [97, 122], true],
  [123, [97, 122], false],
];

describe('is point in range', () => {
  for (const [point, range, expectation] of CASES) {
    test(`${point} in ${JSON.stringify(range)} is ${expectation}`, () => {
      assert.equal(isPointInRange(point, range), expectation);
    });
  }
});
